src/components/drive-container.tsx

```diff
@@ -60,16 +84,64 @@ export default function () {
     function handleNodeClick(item: TreeItem, drive: DriveTreeItem) {
         if (item.type === ItemType.File) {
             openFile(drive.id, item.id);
+        } else {
+            setDrives(drives =>
+                traverseDriveById(drives, drive.id, treeItem => {
+                    if (treeItem.id === item.id) {
+                        return {
+                            ...treeItem,
+                            expanded: !treeItem.expanded,
+                        };
+                    }
+
+                    return treeItem;
+                })
+            );
         }
     }
 
     function addNewFolder(item: TreeItem, drive: DriveTreeItem) {
-        const basePath = item.id.replace(drive.id, '');
-        addFolder(
-            drive.id,
-            `${basePath ? `${basePath}/` : ''}new-folder`, // TODO check if there is a new folder already
-            'New Folder'
+        const normalizedPath = item.id.replace(drive.id, '');
+        const basePath = normalizedPath.split('/').slice(0, -1).join('/');
+
+        const newPath = path.join(
+            basePath,
+            item.label.replace(/\s/g, '-').toLowerCase()
         );
+
+        console.log(newPath);
+
+        addFolder(drive.id, newPath, item.label);
+    }
+
+    function addVirtualNewFolder(item: TreeItem, drive: DriveTreeItem) {
+        const num = Math.floor(Math.random() * 9999) + 1;
```

Comment:
Here I would check `new-folder`, `new-folder-2`, `new-folder-3`, etc to find an id for the new folder. This way it is more deterministic and we know for sure it wont be duplicated.

Reply:
actually this is not longer needed, since wea are creating a virtual folder that has to be renamed by the user. I'll use a fixed name `virtual-new-folder`
aefd83be7c1b0ab035eb0504970fd8ae714f68ab